text/3729-sized-hierarchy.md

```diff
@@ -0,0 +1,2003 @@
+      //~^ error! the trait bound `<T as Add>::Output: Sized` is not satisfied
+      ```
+
+      Relaxing the bounds of an associated type is in effect giving existing
+      parameters a less restrictive bound which is not backwards compatible.
+[^7]: If it was desirable to add a new trait to this RFC's proposed hierarchy
+      then there are four possibilities:
+
+      - Before `Sized`
+        - i.e. `NewSized: Sized: MetaSized: Pointee`
+        - `NewSized` would need to replace `Sized` as a default bound over an
+          edition to avoid this being a breaking change.
+      - Between `Sized` and `MetaSized`
+        - i.e. `Sized: NewSized: MetaSized: Pointee`
+        - Adding bounds on a new trait between `Sized` and `MetaSized` is
+          backwards compatible for some functions (except for on a trait's
+          associated type). An existing function with a `Sized` bound could
+          be relaxed to the new trait without breaking existing code.
+          Existing functions with a `MetaSized` or `Pointee` bound could not.
+      - Between `MetaSized` and `Pointee`
+        - i.e. `Sized: MetaSized: NewSized: Pointee`
+        - Adding bounds on a new trait between `MetaSized` and `Pointee` is
+          backwards compatible for some functions (except for on a trait's
+          associated type). An existing function with a `Sized` or `MetaSized`
+          bound could be relaxed to the new trait without breaking existing code.
+          Existing functions with a `Pointee` bound could not.
+      - After `Pointee`
+        - i.e. `Sized: MetaSized: Pointee: NewSized`
+        - For the same reasons as the traits proposed in this RFC, adding bounds
+          on a new trait after `Pointee` is backwards compatible (except for on a
+          trait's associated type). Any existing function will have stricter
+          bounds than the new trait and so could be relaxed to the new trait
+          without breaking existing code.
+
+## `Sized` bounds
+[sized-bounds]: #sized-bounds
+
+`?Sized` would be made syntactic sugar for a `const MetaSized` bound. A
+`const MetaSized` bound is equivalent to a `?Sized` bound as all values in Rust
+today whose types do not implement `Sized` are valid arguments to
+[`std::mem::size_of_val`][api_size_of_val] and as such have a size which can be
+computed given pointer metadata and knowledge of the type and target platform, and
+therefore will implement `const MetaSized`. As there are currently no
+extern types or other types which would not implement `const MetaSized`,
+every type in Rust today which would satisfy a `?Sized` bound would satisfy
+a `const MetaSized` bound.
+
+**Edition change:** In the current edition,`?Sized` will be syntatic sugar for
+a `const MetaSized` bound. As the `?Trait` syntax is currently accepted for any trait
+but ignored for every trait except `Sized`, `?MetaSized` and `?Pointee` bounds would
```

Comment:
This is being altered by:

- https://github.com/rust-lang/rust/pull/135841

Reply:
Thanks - I'll update the wording here.